examples/hello_part_no_macros/src/main.rs

```diff
@@ -0,0 +1,153 @@
+            period: a653rs::prelude::SystemTime::Infinite,
+            time_capacity: SystemTime::Infinite,
+            entry_point: aperiodic,
+            stack_size: 100_000,
+            base_priority: 1,
+            deadline: Deadline::Soft,
+            name: Name::from_str("aperiodic").unwrap(),
+        };
+        let process_handle = ctx.create_process(process_attributes).unwrap();
+        process_handle.start().unwrap();
+
+        // create periodic process
+        let process_attributes = ProcessAttribute {
+            period: 0.into(),
+            time_capacity: SystemTime::Infinite,
+            entry_point: periodic,
+            stack_size: 100_000,
+            base_priority: 1,
+            deadline: Deadline::Soft,
+            name: Name::from_str("periodic").unwrap(),
+        };
+        let process_handle = ctx.create_process(process_attributes).unwrap();
+        process_handle.start().unwrap();
+    }
+
+    fn warm_start(&self, ctx: &mut a653rs::prelude::StartContext<Hypervisor>) {
+        self.cold_start(ctx)
+    }
+}
+
+/// Entry point for the aperiodic process
+///
+/// This process runs in background mode
+extern "C" fn aperiodic() {
+    info!("Start Aperiodic");
+    for i in 0..i32::MAX {
+        if let SystemTime::Normal(time) = Hypervisor::get_time() {
+            // round the time to an integer value of milliseconds
+            let round = Duration::from_millis(time.as_millis() as u64);
+            info!("{:?}: AP MSG {i}", format_duration(round).to_string());
+        }
+        std::thread::sleep(Duration::from_millis(1))
+    }
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct CustomMessage {
+    msg: String,
+    when: Duration,
+}
+
+extern "C" fn periodic() {
+    let ident = Hypervisor::get_partition_status().identifier;
+    for i in 1..i32::MAX {
+        if let SystemTime::Normal(time) = Hypervisor::get_time() {
+            let round = Duration::from_millis(time.as_millis() as u64);
+            info!("{:?}: P MSG {i}", format_duration(round).to_string());
+        }
+        std::thread::sleep(Duration::from_millis(1));
+
+        if i % 5 == 0 {
+            if ident == 0 {
+                SOURCE_PORT
+                    .get()
+                    .unwrap()
+                    .send_type(CustomMessage {
+                        msg: format!("Sampling MSG {}", i / 5),
+                        when: Hypervisor::get_time().unwrap_duration(),
+                    })
+                    .ok()
+                    .unwrap();
+            } else if ident == 1 {
+                let (valid, data) = DESTINATION_PORT
+                    .get()
+                    .unwrap()
+                    .recv_type::<CustomMessage>()
+                    .ok()
+                    .unwrap();
+
+                info!("Received via Sampling Port: {:?}, valid: {valid:?}", data)
+            }
+
+            Hypervisor::periodic_wait().unwrap();
+        }
+    }
+}
```

Comment:
Duplicate code. You can however add a second binary to the `hello_part` crate and put the process code in `lib`.

Reply:
I strongly oppose this. The idea of examples is to be relatively easy, independent snippets of code that can be easily copied around. I myself get confused quite easily when crate examples are too mixed up with each other, sharing the same code factored out in a dedicated library. I think having no coupling, canonical single-file examples is the best way to go, and DRY should not be a decision driver for examples. On mostly any other kind of code I would agree with you, just not for examples.

Reply:
It is still double the effort for maintaining the example. :-/ Maybe drop the hello_part altogether. We have other examples for the macros like the ping_client and ping_server.

Reply:
I agree with @wucke13 but I also think that the hello world example is way to large for a hello world example.
Maybe we should prune some unnecessary complexity from it